src/plugins/GUITestBase/src/harness/UGUITestBase.cpp

```diff
@@ -25,11 +25,12 @@
 
 namespace U2 {
 
-UGUITestBase::~UGUITestBase() {
-    qDeleteAll(tests);
-    qDeleteAll(preAdditional);
-    qDeleteAll(postAdditionalActions);
-    qDeleteAll(postAdditionalChecks);
+UGUITestBase::UGUITestBase() {
+}
+
+UGUITestBase *UGUITestBase::getInstance() {
+    static UGUITestBase *instance = new UGUITestBase();
```

Comment:
There are no problems to create 2 instances of `UGUITestBase` now - just call this function twice, but the description of this function says "There is only 1 instance of the database per application.". I offer to make it like this:

.h:
```
private:
    static UGUITestBase *instance;
```

.cpp:
```
UGUITestBase*UGUITestBase::instance = nullptr;

UGUITestBase *UGUITestBase::getInstance() {
    if (instance == nullptr) {
        instance = new UGUITestBase();
    }
    
    return instance;
}
```

Reply:
This function will never create the harness twice: the var instance is a local _static_ var.

Reply:
I don't get it, actually...

Reply:
Sorry, I do not understand the question.

Local static vars is compact way to implement singleton pattern in thread safe environment. 
Harness initialization is a part of plugin loading  - so there is no concurrency in initialization and this pattern is valid.

The same way we implement local per-method event and performance counters (about 100+usages across UGENE's code)